flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkScan.java

```diff
@@ -413,6 +415,141 @@ public void testIncrementalRead() throws Exception {
         TestFixtures.SCHEMA);
   }
 
+  @Test
+  public void testIncrementalReadWithTimestampRange() throws Exception {
```

Comment:
we probably should break it down to multiple test methods. one for each scenario

Reply:
I've split the method into smaller parts, could you please take another look?